logstash-core/lib/logstash/logging/slow_logger.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/logging/slow_logger/freq_items"
+
+module LogStash
+  module Logging
+
+    class NullLogger
+      def log(threshold, time, data={})
+      end
+      alias_method :warn, :log
+    end
+
+    class SlowLogger
+
+      attr_reader :logger, :freq_items
+
+      def initialize(name="loggers.slow", params={})
+        @logger     = LogStash::Logging::Logger.new(name)
+        @freq_items = LogStash::Logging::Util::FreqItems.new
+      end
+
+      def log(threshold, time, data={})
```

Comment:
can we also take `Event` by default in this API? We can then log the full event if someone requests it. I would also add a new setting `slowlog.enable.event_logging`. `true` would log the entire original event. By default, it can be `false`.